cmd/configschema/cfgmetadatagen/cfgmetadatagen/cli.go

```diff
@@ -0,0 +1,56 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cfgmetadatagen
+
+import (
+	"fmt"
+	"reflect"
+
+	"go.opentelemetry.io/collector/component"
+	"gopkg.in/yaml.v2"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/cmd/configschema"
+)
+
+// CLI is the entry point for the cfgmetadatagen CLI. Component factories are
+// passed in so it can be used by other distros.
+func CLI(factories component.Factories, sourceDir string, outputDir string) error {
```

Comment:
CLI is still unnecessary exposed publicly. You can move the package under `internal`, but I would rather merge `cmd/configschema/cfgmetadatagen/cfgmetadatagen` package into `cmd/configschema/cfgmetadatagen` and make the function unexported. Probably even put its content into `main` function since I don't believe it significantly improves readability.

Reply:
The reason it's exported is so that other distros (e.g. Splunk Otel Collector) can pass in their own factories as they may have different components. Obviously, they would have to create their own command, but it would be a very small amount of code. Does this sound reasonable, or is there a better way?

Reply:
I see. I was thinking that other distros would use command line tool, but it's good to expose is as a function as well. Can we maybe call it something like `GenerateFiles`? And we still can remove one `cfgmetadatagen` from `cmd/configschema/cfgmetadatagen/cfgmetadatagen` right? :)

Reply:
Sure, I have renamed it. The reason there are two `cfgmetadatagen` directories is because the first one has the main function and `main` package, and the second one has the `cfgmetadatagen` package. Would love to know if there's a better way to do this 🙂

Reply:
Ok. sounds good. Let's keep it as is then